### DISCUSSION TITLE
How to terminate the collector after whole file is harvested?

### DISCUSSION THREAD

**Discussion by graywolf-s1** [open-telemetry/opentelemetry-collector#11648] [Q&A] [answered]
Given a very simple configuration file of

```
receivers:
  filelog:
    include:
      - test-input
    start_at: beginning
    operators:
      - type: json_parser

processors:
  batch:
    send_batch_size: 1
    send_batch_max_size: 1

exporters:
  debug:
    verbosity: normal
    use_internal_logger: false

service:
  pipelines:
    logs:
      receivers:
        - filelog
      processors:
        - batch
      exporters:
        - debug
```

how can I make opentelemetry collector terminate after the whole file is harvested?  This is just for a local testing, so I want to fully process that one file (`test-input`) and stop.  How can I do that?

Thanks

**Comment by dmathieu** [comment, marked as answer]
The collector isn't meant to run as a one-off process. So it can't do that.
You can start the process, and send a SIGTERM signal after a few seconds, that should let the collector finish handling the file, and stop as it handles the signal.

**Reply by graywolf-s1** [reply]
Ah, that is a shame.  For end-to-end local tests it would be useful, and since other software not meant as a one-off process (logstash) can do it, I hoped there is a way.  Nevertheless, I will manage, thank you for the answer.
